src/render/ScreenCreditController.js

```diff
@@ -121,184 +121,34 @@ define([
-            program = dc.currentProgram;
-            gl.bindBuffer(gl.ARRAY_BUFFER, dc.unitQuadBuffer3());
-            gl.vertexAttribPointer(program.vertexPointLocation, 3, gl.FLOAT, false, 0, 0);
-            gl.bindBuffer(gl.ARRAY_BUFFER, dc.unitQuadBuffer());
-            gl.vertexAttribPointer(program.vertexTexCoordLocation, 2, gl.FLOAT, false, 0, 0);
-            gl.enableVertexAttribArray(program.vertexPointLocation);
-            gl.enableVertexAttribArray(program.vertexTexCoordLocation);
+            credit.attributes.color = color;
+            credit.attributes.enableOutline = false;
+            credit.attributes.offset = new Offset(WorldWind.OFFSET_FRACTION, 1, WorldWind.OFFSET_FRACTION, 0.5);
 
-            // Tell the program which texture unit to use.
-            program.loadTextureUnit(gl, gl.TEXTURE0);
-            program.loadModulateColor(gl, false);
-
-            // Turn off depth testing.
-            // tag, 6/17/15: It's not clear why this call was here. It was carried over from WWJ.
-            //gl.disable(WebGLRenderingContext.DEPTH_TEST);
+            this.textCredits.push(credit);
+            this.addRenderable(credit);
         };
 
         // Internal use only. Intentionally not documented.
-        ScreenCreditController.prototype.endDrawingCredits = function (dc) {
-            var gl = dc.currentGlContext,
-                program = dc.currentProgram;
-
-            // Clear the vertex attribute state.
-            gl.disableVertexAttribArray(program.vertexPointLocation);
-            gl.disableVertexAttribArray(program.vertexTexCoordLocation);
+        ScreenCreditController.prototype.doRender = function (dc) {
+            var j = 1;
 
-            // Clear GL bindings.
-            gl.bindBuffer(gl.ARRAY_BUFFER, null);
-            gl.bindTexture(gl.TEXTURE_2D, null);
-
-            // Re-enable depth testing.
-            gl.enable(gl.DEPTH_TEST);
-        };
-
-        // Internal use only. Intentionally not documented.
-        ScreenCreditController.prototype.drawImageCredit = function (dc, creditUrl, x, y) {
-            var imageWidth, imageHeight, scale, activeTexture, gl, program;
-
-            activeTexture = dc.gpuResourceCache.resourceForKey(creditUrl);
-            if (!activeTexture) {
-                dc.gpuResourceCache.retrieveTexture(dc.currentGlContext, creditUrl);
-                return 0;
-            }
-
-            // Scale the image to fit within a constrained size.
-            imageWidth = activeTexture.imageWidth;
-            imageHeight = activeTexture.imageHeight;
-            if (imageWidth <= this.imageCreditSize && this.imageHeight <= this.imageCreditSize) {
-                scale = 1;
-            } else if (imageWidth >= imageHeight) {
-                scale = this.imageCreditSize / imageWidth;
-            } else {
-                scale = this.imageCreditSize / imageHeight;
+            for (var i = 0; i < this.imageCredits.length; i++) {
```

Comment:
Unintentional O(n) loop. Array.length can be, last I checked, a linear-time operation. The pattern we use throughout WorldWind, though it's not obvious at first glance, is as follows:
`for (var i = 0, len = this.imageCredits.length; i < len; i++) {`

Reply:
Interesting. Will do.